test/ResultTest.cpp

```diff
@@ -0,0 +1,585 @@
+    EXPECT_THROW(result.assertThatLimitWasRespected({3, 1}),
+                 ad_utility::Exception);
+    EXPECT_THROW(result.assertThatLimitWasRespected({3, 2}),
+                 ad_utility::Exception);
+  }
+
+  auto createResultsAndApplyAssertion = [&](LimitOffsetClause limitOffset) {
+    std::vector<Result> results;
+    for (auto& generator : getAllSubSplits(idTable)) {
+      Result result{std::move(generator), {}, LocalVocab{}};
+      result.assertThatLimitWasRespected(limitOffset);
+      results.push_back(std::move(result));
+    }
+    return results;
+  };
+
+  for (auto& result : createResultsAndApplyAssertion({})) {
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+  for (auto& result : createResultsAndApplyAssertion({4, 0})) {
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+  for (auto& result : createResultsAndApplyAssertion({4, 1337})) {
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+  for (auto& result : createResultsAndApplyAssertion({42, 0})) {
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+  for (auto& result : createResultsAndApplyAssertion({42, 1337})) {
+    EXPECT_NO_THROW(consumeGenerator(result.idTables()));
+  }
+  for (auto& result : createResultsAndApplyAssertion({3, 0})) {
+    EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
+  }
+  for (auto& result : createResultsAndApplyAssertion({3, 1})) {
+    EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
+  }
+  for (auto& result : createResultsAndApplyAssertion({3, 2})) {
+    EXPECT_THROW(consumeGenerator(result.idTables()), ad_utility::Exception);
+  }
+}
+
+// _____________________________________________________________________________
+TEST(Result,
+     verifyCheckDefinednessDoesThrowIfColumnIsNotDefinedWhenClaimingItIs) {
+  if constexpr (!ad_utility::areExpensiveChecksEnabled) {
+    GTEST_SKIP_("Expensive checks are disabled, skipping test.");
+  }
+  auto correctTable1 = makeIdTableFromVector({{0, 7}, {1, 6}, {2, 5}, {3, 4}});
+  auto correctTable2 =
+      makeIdTableFromVector({{0, Id::makeUndefined()}, {1, 6}, {2, 5}, {3, 4}});
+  auto correctTable3 =
+      makeIdTableFromVector({{0, 7}, {1, 6}, {2, 5}, {3, Id::makeUndefined()}});
+  auto correctTable4 = makeIdTableFromVector({{0, Id::makeUndefined()},
+                                              {1, Id::makeUndefined()},
+                                              {2, Id::makeUndefined()},
+                                              {3, Id::makeUndefined()}});
+  auto wrongTable1 =
+      makeIdTableFromVector({{Id::makeUndefined(), 7}, {1, 6}, {2, 5}, {3, 4}});
+  auto wrongTable2 = makeIdTableFromVector({{Id::makeUndefined(), 7},
+                                            {Id::makeUndefined(), 6},
+                                            {Id::makeUndefined(), 5},
+                                            {Id::makeUndefined(), 4}});
```

Comment:
Bind `Undefined`to a variable, (typically called `U`, then the rows get shorter.